### PR TITLE
Draft to discuss github release for spark-client.

### PR DESCRIPTION
**Pull request by averma-canonical**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: code-checks
+        name: Make sure the code follows conventions and tests correctly
+        run: |
+          make checks
+      - id: build-package
+        name: Build package
+        run: |
+          poetry build
```
Comment (averma-canonical) [inline .github/workflows/release.yaml:40]: assuming this is different from "make build" that we do for the snap.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+name: Create a release
+
+env:
+  BRANCH: release
+  VERSION: 3.3.1
```
Comment (averma-canonical) [inline .github/workflows/release.yaml:5]: How to appropriately regex the BRANCH here?

How to make the VERSION come dynamically?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (deusebio) [reply .github/workflows/release.yaml:40]: yes, it is different. In the SNAP, with build, we build the snap, not the package

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          poetry build
+      - name: Add version to environment vars
+        run: |
+          echo "PROJECT_VERSION=${{ env.VERSION }}" >> $GITHUB_ENV
+      - name: Check if tag version matches project version
+        run: |
+          TAG=$(git describe HEAD --tags --abbrev=0)
+          echo $TAG
+          echo $PROJECT_VERSION
+          if [[ "$TAG" != "v$PROJECT_VERSION" ]]; then exit 1; fi
```
Comment (averma-canonical) [inline .github/workflows/release.yaml:49]: Branch tagging would be a manual process I believe.

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  test:
+    name: Test Release
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+      - name: Download tar.gz file
+        run: |
+          mkdir pkgs
+          # curl download the release and test ?
```
Comment (averma-canonical) [inline .github/workflows/release.yaml:74]: Is post publish test useful?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (averma-canonical) [reply .github/workflows/release.yaml:40]: :+1:

File: .github/workflows/release.yaml (hunk 4 above)
Comment (deusebio) [reply .github/workflows/release.yaml:74]: I would say so!

**Comment by averma-canonical** [comment]
This was useful to discuss and redesign the publish workflow.

**Pull request closed without merging**